### PR TITLE
google_gke: using project_id directly and requiring a value for var.project_id

### PR DESCRIPTION
**Pull request by moz-astults**

### CODE REVIEW

File: google_gke/variables.tf [hunk 1]
```diff
@@ -25,7 +25,6 @@ variable "name" {
 }
 
 variable "project_id" {
-  default     = null
```
Comment (moz-astults) [inline google_gke/variables.tf:28]: If we still want to be able to use the provider's `project_id` (accomplished with the default being `null`) we would need to go through the other areas of this module to replace any `local.project_id` references interpolated as part of a string and use something else. For now, I propose we just require this to be set.

**Comment by moz-astults** [comment]
Thinking about this more, I actually don't think we need the `local.project_id` variable at all if we're going to require `var.project_id` to be set. I added another commit to replace the instances of the local variables.

**Review by cmharlow — APPROVED** [review]
lgtm after discussion of why is this occurring in the first place

**Comment by moz-astults** [comment]
Closing this since we found the root cause - terraform's `trim` function is tricky...

**Pull request closed without merging**